### DISCUSSION TITLE
404 on API Rtsps endpoints

### DISCUSSION THREAD

**Discussion by udatny** [bluenviron/mediamtx#651] [Questions]
## Which version are you using?
docker image
v0.17.6 , initially observed in  0.17.3 

## Describe the issue
on 0.17.3 and on 0.17.6 im encountering 404 when calling the rtsps endpoints:  /v1/rtspsessions/list and /v1/rtspsessions/kick/{id}
is this due to disabled/incomplete rtsps config or is this a bug? or is it a problem on my end?

## Describe how to replicate the issue
take default config
run docker image (with activated api: RTSP_API=yes)

call the api on the mentioned endpoints

**Comment by softko** [comment]
Hello,
I'm using also docker and when I have correct yml, the API endpoint is working. For RTSP session you need to have
encryption set to "no" or "optional" and rtspAddress set. When you have encryption "strict", API path will not work because you can't have RTSP session only RTSPS.
Edit: And for each (rtsp, rtsps) you have different API paths.

**Comment by aler9** [comment]
RTSPS sessions are available only when `encryption` is `optional` or `strict`.

[a comment by an automated account is not shown]
